TankClient.java

```diff
@@ -1,88 +1,106 @@
-		List<Bullets> bullets = new ArrayList<Bullets>();
-		List<Tree> trees = new ArrayList<Tree>();
-		List<CommonWall> homeWall = new ArrayList<CommonWall>();
-		List<CommonWall> otherWall = new ArrayList<CommonWall>();
-		List<MetalWall> metalWall = new ArrayList<MetalWall>();
-		private static final String TIMES_NEW_ROMAN = "Times New Roman";
-
-		public void update(Graphics graphics) {
-
-			screenImage = this.createImage(FRAM_WIDTH, FRAM_LENGTH);
-
-			Graphics gps = screenImage.getGraphics();
-			Color color = gps.getColor();
-			gps.setColor(Color.GRAY);
-			gps.fillRect(0, 0, FRAM_WIDTH, FRAM_LENGTH);
-			gps.setColor(color);
-			framPaint(gps);
-			graphics.drawImage(screenImage, 0, 0, null);
+import java.util.ArrayList;
+import java.awt.*;
+import java.awt.event.*;
+import java.util.List;
+import javax.swing.JOptionPane;
+
+public class TankClient extends Frame implements ActionListener {
+
+	/**
+	 *
+	 */
+	private static final long serialVersionUID = 1L;
+	public static final int FRAM_WIDTH = 800;
+	public static final int FRAM_LENGTH = 600;
+	private boolean printable = true;
+	MenuBar jmb = null;
+	Menu jm1 = null;
+	Menu jm2 = null;
+	Menu jm3 = null;
+	Menu jm4 = null;
+	Menu jm5 = null;
+	MenuItem jmi1 = null;
+    MenuItem jmi2 = null;
+    MenuItem jmi3 = null;
+    MenuItem jmi4 = null;
+    MenuItem jmi5 = null;
+    MenuItem jmi6 = null;
+    MenuItem jmi7 = null;
+    MenuItem jmi8 = null;
+    MenuItem jmi9 = null;
+    MenuItem jmi10 = null;
+    MenuItem jmi11 = null;
+	Image screenImage = null;
+
+	Tank homeTank = new Tank(300, 560, true, Direction.STOP, this,1);
+	Tank homeTank2 = new Tank(449, 560,true,Direction.STOP,this,2);
+	Boolean player2 = false;
+	GetBlood blood = new GetBlood();
+	Home home = new Home(373, 557, this);
+	Boolean win = false;
+	Boolean lose = false;
+	List<River> theRiver = new ArrayList<River>();
+	List<Tank> tanks = new ArrayList<Tank>();
+	List<BombTank> bombTanks = new ArrayList<BombTank>();
+	List<Bullets> bullets = new ArrayList<Bullets>();
+	List<Tree> trees = new ArrayList<Tree>();
+	List<CommonWall> homeWall = new ArrayList<CommonWall>();
+	List<CommonWall> otherWall = new ArrayList<CommonWall>();
+	List<MetalWall> metalWall = new ArrayList<MetalWall>();
+	private static final String TIMES_NEW_ROMAN = "Times New Roman";
+
+	public void update(Graphics graphics) {
+
+		screenImage = this.createImage(FRAM_WIDTH, FRAM_LENGTH);
+
+		Graphics gps = screenImage.getGraphics();
+		Color color = gps.getColor();
+		gps.setColor(Color.GRAY);
+		gps.fillRect(0, 0, FRAM_WIDTH, FRAM_LENGTH);
+		gps.setColor(color);
+		framPaint(gps);
+		graphics.drawImage(screenImage, 0, 0, null);
+	}
+
+	public void framPaint(Graphics graphics) {
+
+		Color c = graphics.getColor();
+		graphics.setColor(Color.green);
+
+		Font f1 = graphics.getFont();
+		graphics.setFont(new Font(TIMES_NEW_ROMAN, Font.BOLD, 20));
+		if(!player2)graphics.drawString("Tanks left in the field: ", 200, 70);

```

Comment:
Yra daug if metodų, kurie apsunkina kodą, padaro jį mažiau skaitomų. Kiekvieną iš if'ų kurie yra "framPaint" metode galima supaprastinti parašant viską į vieną eilutę. Taip bus geresnis kodo skaitomumas ir bus daugiau suprantamas.
pvz:  graphics.drawString("Tanks left in the field: ", !player2 ? 200 : 100, 70);